### DISCUSSION TITLE
Issue Installing Git: Lesson 6 | Foundry Simple Storage 6:32:12

### DISCUSSION THREAD

**Discussion by beeski00** [Cyfrin/foundry-full-course-cu#784] [Q&A] [answered]
Hey All,

Whenever I go to install git, I receive the following error from my mac: "Can’t install the software because it is not currently available from the Software Update server."

I search the web and saw several answers regarding a similar issue while getting windows on mac. Has anyone come across this issue and know how to resolve? 

Is git an essential for continuing the course? 

Thank you in Advanced.

Beeski

**Comment by elec7rik** [comment]
hi beeski, what method are you using to install git? and yes, i would say git is essential for the course otherwise you might face issues later on that will push you behind so it would be better to get done with it right away. 
i am not on mac so i can't show you the exact procedure but this [atlassian article](https://www.atlassian.com/git/tutorials/install-git) may be useful. 
first, check with ```git --version``` if you already have it.
if you don't or if you want to upgrade to a newer version, you can follow the 4 steps mentioned under the heading **Git for Mac Installer**
it should be fairly simple, feel free to share if you still get any errors.

**Reply by beeski00** [reply]
Hey Vicky, thank you for your reply. 

I attempted to install using a few ways: first was typing in git into my studio's terminal. Which it then prompts me to install and read through some terms and agreements. Once I click agree, it begins the install and then aborts with the error mentioned above. 

the second way I tried was through homebrew where it prompts me to paste code into mac's terminal. Same process occurred where it error'd out with the message above. 

I then tried to download .pkg installer form Homebrews github release and rec'd the following error when I opened the package - Command Line Tools are missing. "You must install the Command Line Tools (CLT) before installing Homebrew by running `xcode-select --install` from a Terminal. " I inserted this code into my terminal and same error :( .

**Reply by beeski00** [reply, marked as answer]
Updating my mac - hopefully that helps

**Reply by elec7rik** [reply]
seems like you don't have xcode installed so the CLT will not work. 
1. try to download xcode from the app store. best case, it should fix everything.
2. if it doesn't, refer to [this](https://developer.apple.com/forums/thread/660641) thread. a lot of people seemed to have faced this issue. some of them fixed it by downloading xcode from the app store and others had to download a file such as ```[Command_Line_Tools_for_Xcode_12_GM_seed.dmg](https://download.developer.apple.com/Developer_Tools/Command_Line_Tools_for_Xcode_12_GM_seed/Command_Line_Tools_for_Xcode_12_GM_seed.dmg)```
3. **note** that this link is for xcode12, so download the file according to the version that you need. [this](https://idmsa.apple.com/IDMSWebAuth/signin.html?path=%2Fdownload%2Fall%2F&appIdKey=891bd3417a7776362562d2197f89480a8547b108fd934911bcbea0110d07f757&rv=0) page should have the version that you would need.

i hope this helps.

**Reply by elec7rik** [reply]
oh yes it should. i was assuming your system to be updated, my bad. but if the issue still persists try the points mentioned above.

**Reply by beeski00** [reply]
Thank you for helping me talk it out lol. After, updating my mac - its working :)
